src/commands/secrets/upload.ts

```diff
@@ -0,0 +1,98 @@
+import { Flags } from '@oclif/core';
+import fs from 'fs-extra';
+import yaml from 'js-yaml';
+import path from 'path';
+import tmp from 'tmp';
+import untildify from 'untildify';
+import { Dictionary } from '../..';
+import AccountUtils from '../../architect/account/account.utils';
+import { EnvironmentUtils } from '../../architect/environment/environment.utils';
+import Command from '../../base-command';
+import { SecretsDict } from '../../dependency-manager/secrets/type';
+
+tmp.setGracefulCleanup();
+
+interface Secret {
+  scope: string;
+  key: string;
+  value: string;
+}
+
+export default class SecretsUpload extends Command {
+  async auth_required(): Promise<boolean> {
+    return false;
+  }
+
+  static description = 'Upload secrets from a file to an account or an environment';
+  static aliases = ['secrets:set'];
+
+  static flags = {
+    ...Command.flags,
+    ...AccountUtils.flags,
+    ...EnvironmentUtils.flags,
+    override: Flags.boolean({
+      description: 'Allow override of existing secrets',
+      default: false,
+    }),
+  };
+
+  static args = [{
+    name: 'secrets_file',
+    description: 'Secrets file to be uploaded',
+    required: true,
+  }];
+
+  async run(): Promise<void> {
+    const { flags, args } = await this.parse(SecretsUpload);
+    const account = await AccountUtils.getAccount(this.app, flags.account);
+
+    const { data: user } = await this.app.api.get('/users/me');
```

Comment:
All duplicated logic should be moved to a single file. We could have a UserUtil with a function that returns membership type for instance.